webui/react/src/hooks/useModal/useJupyterLabModal.tsx

```diff
@@ -71,11 +66,13 @@ interface FullConfigProps {
 
 const MonacoEditor = React.lazy(() => import('components/MonacoEditor'));
 
-const JupyterLabModal: React.FC<JupyterLabModalProps> = (
-  { visible = false, onLaunch, ...props }: JupyterLabModalProps,
-) => {
+const useJupyterLabModal = (): ModalHooks => {
+  const { modalClose, modalOpen: openOrUpdate, modalRef } = useModal();
+
   const [ showFullConfig, setShowFullConfig ] = useState(false);
   const [ config, setConfig ] = useState<string | undefined>();
+  const [ previousConfig, setPreviousConfig ] = useState<string | undefined>();
+  const [ previousShowConfig, setPreviousShowConfig ] = useState<boolean>(false);
```

Comment:
**suggestion:** if you want to compare `showFullConfig` to the previous value it was set to, the `usePrevious` hook might be better to use instead of setting up a separate state to keep track of this.
```
const prevShowFullConfig = usePrevious(showFullConfig, undefined);
```
the default value is set to `undefined` in case you also want to detect changes into being set to `false` but it could also just be `false` if you don't care about that.

Caution: the `usePrevious` has to be defined **after** the definition of the state. e.g. `prevShowFullConfig` has to be defined after `showFullConfig`.

Reply:
actually, it looks like even `previousConfig` might be another great candidate for `usePrevious` also.